### PR TITLE
Create foodlist

### PR DESCRIPTION
**Pull request by FooChiHen**

### CODE REVIEW

File: src/database/foodlist.go [hunk 1]
```diff
@@ -3,8 +3,50 @@ package database
...
+				LEFT JOIN foodloop.tag t 
+				ON t.tagID = ftt.tagID 
+				WHERE t.tagName 
+				IN ('`+concatTags+`')
+				) as ff
+			GROUP BY ff.foodName, ff.descriptions
+			) as fff
+		WHERE fff.count = $1
+		`,
+		2,
```
Comment (FooChiHen) [inline src/database/foodlist.go]: this value can be changed for however many tag matches we want. higher value more strigent

File: src/database/foodlist.go [hunk 2]
```diff
@@ -3,8 +3,50 @@ package database
...
+	)
+	if err != nil {
+		fmt.Println(err)
+	}
```
Comment (mahe993) [inline src/database/foodlist.go]: can we `return nil, err` here as well so client response can react accordingly

File: src/database/foodlist.go [hunk 3]
```diff
@@ -3,8 +3,50 @@ package database
...
+	res := []models.Food{}
+	for rows.Next() {
+		var food models.Food
+		if err := rows.Scan(
+			&food.FoodName,
+			&food.Descriptions,
+		); err != nil {
+			fmt.Println(err)
```
Comment (mahe993) [inline src/database/foodlist.go]: same here please

File: src/services/foodlist.go [hunk 4]
```diff
@@ -1,16 +1,52 @@
...
+	}
+	var resp postResp
+	err = json.Unmarshal(body, &resp)
+	if err != nil {
+		render.Status(r, http.StatusBadRequest)
+		render.JSON(w, r, err.Error())
+		return
+	}
+	tags := cleanQuery(resp.Query)
+	foodlist := database.GetTagsID(tags)
```
Comment (mahe993) [inline src/services/foodlist.go]: render error here too please

**Review by mahe993 — APPROVED** [review]
Just the error handling and we are good! 👍

File: src/database/foodlist.go [hunk 5]
```diff
@@ -24,33 +24,80 @@ func GetTagsID(tags []string) []models.Food {
...
+
+	if _, err := db.Exec(
+		`
+		INSERT INTO foodloop.foodlist(foodlistID)
+		VALUES($1)
+		`,
+		currID+1,
+	); err != nil {
+		return err
+	}
```
Comment (mahe993) [inline src/database/foodlist.go]: Primary key IDs should be auto incrementing, we shouldnt have to search for ID and add the latest. This looks like a primary key if I am not mistaken?

File: src/database/foodlist.go [hunk 6]
```diff
@@ -24,33 +24,80 @@ func GetTagsID(tags []string) []models.Food {
...
-			fmt.Println(err)
+			return nil, err
 		}
 		res = append(res, food)
 	}
-	return res
+	return res, nil
+}
+
+func InsertFoodlist(id int, list []models.Food) error {
```
Comment (mahe993) [inline src/database/foodlist.go]: this should take in the userID and foodlistID

**Review by mahe993 — DISMISSED** [review]
comments on primary ID

File: src/services/foodlist.go [hunk 7]
```diff
@@ -1,16 +1,71 @@
...
+		render.Status(r, http.StatusBadRequest)
+		render.JSON(w, r, "we couldn't generate a list for that query :(")
+		return
+	}
+	if err := database.InsertFoodlist(userID, foodlist, resp.Day, resp.Time, resp.Title); err != nil {
+		render.Status(r, http.StatusBadRequest)
+		render.JSON(w, r, err.Error())
+		return
+	}
+	render.JSON(w, r, foodlist)
```
Comment (mahe993) [inline src/services/foodlist.go:65]: can we sort foodlist.food in asc by the food idx before response?

Comment (FooChiHen) [reply src/services/foodlist.go:65]: done

**Review by chae-seong — APPROVED** [review]
LGTM

**Pull request merged**